File: Python3/0013. Roman to Integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        # num = {'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        # res = 0
        # temp = 0
        # for i in reversed(s):
        #     if(num[i]>=temp):
        #         res += num[i]
        #     else:
        #         res -= num[i]
        #     temp = num[i]
        # return res
        num = {'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        temp = 0
        res = 0
        for i in s:
            if(num[i]>temp):
                res -= temp
            else:
                res += temp
            temp = num[i]
        res += num[s[-1]]
        return res

File: C++/0013. Roman to Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char, int> num = {{'I',1}, {'V',5}, {'X',10}, {'L',50}, {'C',100}, {'D',500}, {'M',1000}};
        int res = 0;
        int temp = 0;
        for(char c:s) {
            if(temp < num[c]) {
                res -= temp;
            } else {
                res += temp;
            }
            temp = num[c];
        }
        return res + temp;
    }
};
